routing/router.go

```diff
@@ -1327,6 +1335,16 @@ func (r *ChannelRouter) processUpdate(msg interface{},
 				"graph: %v", msg.PubKeyBytes, err)
 		}
 
+		pk, err := msg.PubKey()
+		if err != nil {
+			return err
+		}
+
+		// If we currently have any connection requests pending for
+		// this node then update the requests if the node addresses
+		// have changed.
+		r.cfg.UpdateConnRequests(pk, msg.Addresses)
```

Comment:
Not sure if `router` should be handling this as it's already saving the node info to db. Some other service could just read the info from db and apply the necessary changes there.

Reply:
See my comment below

Reply:
related to my comment below, yes sure another service could read from the DB but the issue is that the other service would not know _when_ to refresh from the db

Reply:
Though we haven't strictly defined the overall architecture, `routing` package should not be worrying about `net.Addr`. Its main purpose is to route HTLCs over the link layer. If other services don't know when to refresh the db, we could apply a publish-subscribe pattern, or, in this case, might just periodically refresh the db.

Reply:
cool I have moved this to the gossiper package.